topi/python/topi/vision/rcnn/roi_align.py

```diff
@@ -0,0 +1,91 @@
+# pylint: disable=invalid-name
+"""Roi align operator"""
+import tvm
+from ...util import get_const_tuple
+
+@tvm.target.generic_func
+def roi_align(data, rois, pooled_size, spatial_scale, sample_ratio=-1):
+    """ROI align operator in NCHW layout.
+
+    Parameters
+    ----------
+    data : tvm.Tensor
+        4-D with shape [batch, channel, height, width]
+
+    rois : tvm.Tensor
+        2-D with shape [num_roi, 5]. The last dimension should be in format of
+        [batch_index, w_start, h_start, w_end, h_end]
+
+    pooled_size : int or list/tuple of two ints
+        output size, or [out_height, out_width]
+
+    spatial_scale : float
+        Ratio of input feature map height (or w) to raw image height (or w). Equals the reciprocal
+        of total stride in convolutional layers, which should be in range (0.0, 1.0]
+
+    sample_ratio : int
+        Optional sampling ratio of ROI align, using adaptive size by default.
+
+    Returns
+    -------
+    output : tvm.Tensor
+        4-D with shape [num_roi, channel, pooled_size, pooled_size]
+    """
+    dtype = rois.dtype
+    _, channel, height, width = get_const_tuple(data.shape)
+    num_roi, _ = get_const_tuple(rois.shape)
+
+    if isinstance(pooled_size, int):
+        pooled_size_h = pooled_size_w = pooled_size
+    else:
+        pooled_size_h, pooled_size_w = pooled_size
+
+    def _bilinear(i, c, y, x):
+        outside = tvm.any(y < -1.0, x < -1.0, y > height, x > width)
+        y = tvm.max(y, 0.0)
+        x = tvm.max(x, 0.0)
+        y_low = y.astype('int32')
+        x_low = x.astype('int32')
+        y_high = tvm.min(y_low + 1, height - 1)
+        x_high = tvm.min(x_low + 1, width - 1)
+        y_lerp = y - y_low
+        x_lerp = x - x_low
+        bottom = x_lerp * data[i, c, y_high, x_high] + (1 - x_lerp) * data[i, c, y_high, x_low]
+        top = x_lerp * data[i, c, y_low, x_high] + (1 - x_lerp) * data[i, c, y_low, x_low]
+        val = y_lerp * bottom + (1 - y_lerp) * top
+        return tvm.select(outside, 0.0, val)
+
```

Comment:
it would be nice if we could factor biliner sampling logic that we already have in topi [bilinear upsamling](https://github.com/dmlc/tvm/blob/master/topi/include/topi/image/resize.h#L172). Might be useful for other ops as well.